terraform/main.tf

```diff
@@ -0,0 +1,499 @@
+          Service = [
+            "organizations.amazonaws.com"
+          ]
+        }
+      },
+      {
+        Effect = "Allow",
+        Action = "sts:AssumeRole",
+        Principal = {
+          AWS = "${resource.aws_iam_role.data_transformer_lambda_exec_role.arn}"
+        }
+      }
+    ]
+  })
+}
+
+# Attached AWS managed AWSOrganizationsReadOnlyAccess policy to the Query Org Accounts Role
+resource "aws_iam_role_policy_attachment" "query_org_accounts_access" {
+  provider   = aws.management
+  role       = aws_iam_role.query_org_accounts.name
+  policy_arn = "arn:aws:iam::aws:policy/AWSOrganizationsReadOnlyAccess"
+}
+
+# CloudGuardDataTransformer Lambda role
+resource "aws_iam_role" "data_transformer_lambda_exec_role" {
+  name = "CloudGuardDataTransformerExecutionRole"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = "sts:AssumeRole",
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+# Assume role right for the org list account
+resource "aws_iam_policy" "cloudguard_dashboard_data_transformer_policies" {
+  name = "CloudGuardDataTransformerAssumeRole"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow",
+        Action   = "sts:AssumeRole",
+        Resource = "${resource.aws_iam_role.query_org_accounts.arn}"
+      }
+    ]
+  })
+}
+
+# Attach policies
+resource "aws_iam_role_policy_attachment" "data_transformer_lambda_assumerole" {
+  role       = aws_iam_role.data_transformer_lambda_exec_role.name
+  policy_arn = aws_iam_policy.cloudguard_dashboard_data_transformer_policies.arn
+}
+
+resource "aws_iam_role_policy_attachment" "data_transformer_basic_execution_policy" {
+  role       = aws_iam_role.data_transformer_lambda_exec_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "data_transformer_parameter_store_readonly_policy" {
+  role       = aws_iam_role.data_transformer_lambda_exec_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess"
+}
+
+# Attach the LambdaPermissions IAM policy to DataTransformer Lambda Execution Role
+resource "aws_iam_role_policy_attachment" "data_transformer_lambda_access" {
+  role       = aws_iam_role.data_transformer_lambda_exec_role.name
+  policy_arn = aws_iam_policy.cloudguard_dashboard_lambda_policies.arn
+}
+
+resource "aws_lambda_function" "data_transformer_lambda" {
+  function_name = "CloudGuardDataTransformer"
+  description   = "Transforms raw CloudGuard data for consumption by QuickSight"
+
+  s3_bucket = aws_s3_bucket.cloudguard_dashboard_lambda_bucket.id
+  s3_key    = aws_s3_bucket_object.cloudguard_data_transformer.key
+
+  runtime                        = "nodejs14.x"
+  handler                        = "src/index.index"
+  timeout                        = 180
```

Comment:
Is it possible that this timeout is part of the cold start issue?

Reply:
Sadly it's not related :(
I've tried to increase it but it's not the lambda that timeout, it's the stream that receive the end signal unexpectedly.
Indeed the file is written in s3 meaning the on.('end') part of the code is played.
But this file produced weight ~70kb (expected: 240kb) and contains around (it's not consistent) 1000 line (expected: 4000)
I don't get why it receives it with only roughly the fourth of the input file size and it might deserve a AWS ticket because this issue only appears on lambda environment.